File: a1/rm_ws.c
//Yunfan Li
//liyunf@onid.oregonstate.edu
//CS344-001
//Assignment #1
//No references
#include <stdio.h>
#include <stdlib.h>
#include <string.h>
#include <unistd.h>
#define MAX_NUM 1024

void handle_b(char *s);
void handle_i(char *s);
void handle_e(char *s);

//Delete all the spaces and tabs at front
void handle_b(char *s)
{
	while ((s[0] == 32) || (s[0] == 9)) {
		int i = 1;
		for (; i <= strlen(s); i++)
			s[i-1] = s[i];
	}
}

//Delete all the spaces and tabs at middle
void handle_i(char *s)
{
	int i = 0;
	int start, end;
	while ((s[i] == 32) || (s[i] == 9))
		i++;
	start = i;
	i = strlen(s);
	while ((s[i] == 32) || (s[i] == 9) || (s[i] == '\n'))
		i--;
	end = i;
	for (i = start; i <= end; i++)
		while ((s[i] == 32) || (s[i] == 9)) {
			int j = i+1;
			for(; j <= strlen(s); j++)
				s[j-1] = s[j];
			end--;
		}
}

//Delete all the spaces and tabs at end
void handle_e(char *s)
{
	int i = strlen(s)-1;
	while ((s[i] == 32) || (s[i] == 9) || (s[i] == '\n')) {
        s[i] = '\0';
		i--;
	}
    s[i+1] = '\n';
}

int main(int argc, char **argv)
{	
	char buf[MAX_NUM], opt[3], tmp;
	
	memset(buf, -1, MAX_NUM);
	memset(opt, 0, 3);

	if(argc > 1)
		while ((tmp = getopt(argc, argv, "bei")) != -1)
			switch (tmp) {
			case 'b':
				opt[0] = 1;
				break;
			case 'i':
				opt[1] = 1;
				break;
			case 'e':
				opt[2] = 1;
				break;
			}
	
	while (fgets(buf, MAX_NUM, stdin) != NULL) {
		if (opt[0] == 1)
			handle_b(buf);
		if (opt[1] == 1)
			handle_i(buf);
		if (opt[2] == 1)
			handle_e(buf);
		printf("%s", buf);
	}

	return 0;
}

File: a1/rm_ws.py
#! /usr/bin/env python2.7
#Yunfan Li
#liyunf@onid.oregonstate.edu
#CS344-001
#Assignment #1
#No references
import sys, os, getopt

flag_b = flag_i = flag_e = 0

opts, args = getopt.getopt(sys.argv[1:], 'bie')

for opt in opts:
    if opt[0] == '-b':
        flag_b = 1
    elif opt[0] == '-i':
        flag_i = 1
    elif opt[0] == '-e':
        flag_e = 1

while 1:
    line = sys.stdin.readline()
    
    if line:
        if flag_b:
            i = 0
            while ord(line[i]) == 32 or ord(line[i]) == 9:
                i += 1
            line = line[i:len(line)]
        if flag_i:
            start = 0
            end = len(line)-1
            while ord(line[start]) == 32 or ord(line[start]) == 9:
                start += 1
            while ord(line[end]) == 32 or ord(line[end]) == 9:
                end -= 1
            i = start
            while i >= start and i <=end:
                while ord(line[i]) == 32 or ord(line[i]) == 9:
                    line = line[0:i] + line[i+1:len(line)]
                    end = len(line)-1
                i += 1
        if flag_e:
            end = len(line)-1
            while ord(line[end]) == 32 or ord(line[end]) == 9:
                end -= 1
            line = line[0:end+1]    
        print line,
    else:
        break
